### PR TITLE
8324630: C1: Canonicalizer::do_LookupSwitch doesn't break the loop when the successor is found

### PR DESCRIPTION
**Pull request by D-D-H**
Hi,

Please review the small change that breaks the loop in Canonicalizer::do_LookupSwitch if the successor is found.

The keys of LookupSwitch are sorted, so there is no need to continue the loop once matched.

Thanks.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8324630](https://bugs.openjdk.org/browse/JDK-8324630): C1: Canonicalizer::do_LookupSwitch doesn't break the loop when the successor is found (**Enhancement** - P4)

### Reviewers
 * [Dean Long](https://openjdk.org/census#dlong) (@dean-long - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17553/head:pull/17553` \
`$ git checkout pull/17553`

Update a local copy of the PR: \
`$ git checkout pull/17553` \
`$ git pull https://git.openjdk.org/jdk.git pull/17553/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17553`

View PR using the GUI difftool: \
`$ git pr show -t 17553`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17553.diff">https://git.openjdk.org/jdk/pull/17553.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17553#issuecomment-1908148565)

### CODE REVIEW

File: src/hotspot/share/c1/c1_Canonicalizer.cpp
Comment (dean-long) [inline src/hotspot/share/c1/c1_Canonicalizer.cpp:848]: Shouldn't we also break when `v < x->key_at(i) `, meaning no key will match?  Maybe we should consider binary search?

Comment (D-D-H) [reply src/hotspot/share/c1/c1_Canonicalizer.cpp:852]: Updated: break when `v < x->key_at(i)`

> Maybe we should consider binary search?
From the perspective of pursuing performance, binary search can be considered. If you agree, I can change to using binary search here.

Comment (dean-long) [reply src/hotspot/share/c1/c1_Canonicalizer.cpp:852]: Yes, I'd like to see how it looks.

Comment (D-D-H) [reply src/hotspot/share/c1/c1_Canonicalizer.cpp:852]: Updated.

File: src/hotspot/share/c1/c1_Canonicalizer.cpp
Comment (dean-long) [inline src/hotspot/share/c1/c1_Canonicalizer.cpp:848]: Isn't this the same as
`int mid = (low + high) >> 1;`

Comment (D-D-H) [reply src/hotspot/share/c1/c1_Canonicalizer.cpp:848]: `low + ((high - low) >> 1)`  can avoid integer overflow

Comment (dean-long) [reply src/hotspot/share/c1/c1_Canonicalizer.cpp:848]: Good point.  Integer overflow is a real concern.

**Review by dean-long — APPROVED** [review]
/reviewers 2

**Comment by D-D-H** [comment]
Could I have a second review?

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by D-D-H** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**
